File: Python/168.Excel Sheet Column Title.py
#返回数字对应的字母串,1->A..26->Z...27->AA,28->AB...
class Solution(object):
    def convertToTitle(self, n):
        res = ''
        while n > 0:
            tmp = n
            n = (tmp-1)//26
            res += chr(65+(tmp-1)%26)
        return res[::-1]

s = Solution()
print(s.convertToTitle(128))

File: C++/Math/168.Excel Sheet Column Title.cpp
/*
整数转换成字符串。
	1 -> A
    2 -> B
    3 -> C
    ...
    26 -> Z
    27 -> AA
    28 -> AB 
*/

#include <iostream>
#include <cstring>
using namespace std;

class Solution {
public:
	string convertToTitle(int n) {
        if(n<=0)	return "";
        string result = "";
        while(n)
        {
        	result = char(--n%26+'A')+result;
        	n /= 26;
        }
        return result;
    }
};
int main()
{
	Solution s;
	cout<<s.convertToTitle(180)<<endl;
}

